Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/PerfLib/CategorySample.cs

```diff
@@ -0,0 +1,258 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Runtime.InteropServices;
+
+    internal class CategorySample
+    {
+        public Dictionary<int, CounterDefinitionSample> CounterTable;
```

Comment:
That's an option, but that would mean an extra constructor in the production implementation that we don't need (and it would make zero sense semantically, too). My judgement call is that we're ok without the readonly keyword here.